app/decorators/employee_decorator.rb

```diff
@@ -0,0 +1,32 @@
+class EmployeeDecorator < Draper::Decorator
+  delegate_all
+
+  def initialize(employee)
+    @employee = employee
+  end
+
+  SALARY_INCREASE_RATE = 0.012
+
+  def calculate_salary
+    unless @employee.positions.empty?
+      base_salary = @employee.positions.last.salary
+      years_of_service = years_of_employee
+
+      increased_salary = (base_salary * (1 + SALARY_INCREASE_RATE)**years_of_service).to_i
+    end
+  end
+
+  def years_of_employee
+    total_days = 0
+    @employee.position_histories.each do |position|
+      if position.finished_on.present?
```

Comment:
Use the ternary operator when you have two blocks that start with the same logic.
`total_days += p.finished_on ? (p.finished_on - p.started_on).to_i  : (DateTime.now - p.started_on).to_i`
Also, in order to have it more readable, create a method `duration` for the position, and then you have: 
`total_days += position.duration`